### ISSUE TITLE
Panic when length of comment >= input_buffer_size (on v3 branch)

### ISSUE THREAD

**Issue by logan** [go-yaml/yaml#469]
Here's a simple reproduction as a header comment, but I first spotted this on a footer comment:

```go
	doc := make([]byte, 512*3)
	for i := range doc {
		doc[i] = ' '
	}
	doc[0] = '#'
	yaml.NewDecoder(bytes.NewReader(doc)).Decode(&yaml.Node{})
```

**Comment by niemeyer** [comment]
Thanks for the report. I'll get that fixed.

**Comment by niemeyer** [comment]
That's fixed in 0932294.
